code/game/objects/items/food/frozen.dm

```diff
@@ -117,6 +121,10 @@
 	tastes = list("ice" = 1, "water" = 1, "limes" = 5)
 	foodtypes = FRUIT
 
+/obj/item/food/snowcones/lime/make_edible()
+	. = ..()
+	AddComponent(/datum/component/ice_cream_holder)
```

Comment:
Why are all of these necessary? Surely you only need to do this on the parent (`/obj/item/food/snowcones`)?

Reply:
I tried that, however it didn't seem to pass down in my testing.

Reply:
Well that doesn't make any sense, because I would wager all of these child procs are doing nothing

Reply:
I thought this was what fixed it, however it appears not. I now know what the actual problem is, I just have to work out why.